common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -39,17 +86,735 @@ class NativeAuthMsalController : BaseNativeAuthController() {
     companion object {
         private val TAG = NativeAuthMsalController::class.java.simpleName
     }
-    
+
+    /**
+     * Acts as the starting point for the Sign In flow.
+     * Starts off with the sign in initiate call, and returns any error results if produced.
+     * If successful, continues ot the sign in challenge call, again returning any error results if produced.
+     * If the challenge endpoint returns a password required result,
+     * a call to the token endpoint will be made with the password provided in the SignInStartCommandParameters,
+     * again returning any error results, or returning the token if the call succeeds.
+     * If a token is acquired, it is also cached locally as part of this flow.
+     */
+    fun signInStart(parameters: SignInStartCommandParameters): SignInStartCommandResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signInStart")
```

Comment:
Could we log if it's sign in with password or with code?

Reply:
I guess that could show up later in the logs anyway. But perhaps clarity here would be useful

Reply:
These logs are just for logging the current method. For this reason, the information in parameters is not logged.

Reply:
Doesn't have to be in this exact log method, just wondering if we should add a new log to tell us whether we are going password or code path.